src/Altinn.Apps/AppTemplates/AspNet/Altinn.App.Api/Controllers/DataController.cs

```diff
@@ -600,5 +613,89 @@ private static bool IsValidContributer(DataType dataType, ClaimsPrincipal user)
+        {
+            errorMessage = string.Empty;
+
+            if (!Request.Headers.ContainsKey("Content-Disposition"))
+            {
+                errorMessage = "Conent-Disposition header containing 'filename' must be included in request.";
+                return false;
+            }
+
+            Request.Headers.TryGetValue("Content-Disposition", out StringValues headerValues);
+            string filename = GetFilenameFromContentDisposition(headerValues.ToString());
+
+            if (string.IsNullOrEmpty(filename))
+            {
+                errorMessage = "Content-Disposition header must contain 'filename'.";
+                return false;
+            }
+
+            string[] splitFilename = filename.Split('.');
+
+            if (splitFilename.Count() < 2)
+            {
+                errorMessage = $"Invalid format for filename: {filename}. Filename is expected to end with '.{{filetype}}'.";
+                return false;
+            }
+
+            // no restrictions on data type
+            if (dataType.AllowedContentTypes == null || dataType.AllowedContentTypes.Count == 0)
+            {
+                return true;
+            }
+
+            string filetype = splitFilename[splitFilename.Length - 1];
+            string mimeType = MimeTypeMap.GetMimeType(filetype);
+
+            if (!Request.Headers.ContainsKey("Content-Type"))
+            {
+                errorMessage = "Content-Type header must be included in request.";
+                return false;
+            }
+
+            // Verify that file mime type matches content type in request
+            Request.Headers.TryGetValue("Content-Type", out StringValues contentType);
+            if (!contentType.Equals("application/octet-stream") && !mimeType.Equals(contentType, StringComparison.InvariantCultureIgnoreCase))
+            {
+                errorMessage = $"Content type header {contentType} does not match mime type {mimeType} for uploaded file. Please fix header or upload another file.";
+                return false;
+            }
+
+            // Verify that file mime type is an allowed content-type
+            if (!dataType.AllowedContentTypes.Contains(mimeType, StringComparer.InvariantCultureIgnoreCase) && !dataType.AllowedContentTypes.Contains("application/octet-stream"))
+            {
+                errorMessage = $"Invalid content type: {mimeType}. Please try another file. Permitted content types include: {String.Join(", ", dataType.AllowedContentTypes)}";
+                return false;
+            }
+
+            return true;
+        }
+
+        private string GetFilenameFromContentDisposition(string contentdisposition)
+        {
+            string filename = string.Empty;
+            string keyWord = "filename=";
+
+            if (!contentdisposition.Contains(keyWord, StringComparison.InvariantCultureIgnoreCase))
+            {
+                return filename;
```

Comment:
As this is always returning string.Empty it could be easier to read if that was the statement.

Reply:
done